pkg/loader/executor.go

```diff
@@ -269,10 +277,54 @@ func (e *executor) splitExecDML(ctx context.Context, dmls []*DML, exec func(dmls
 	return errors.Trace(errg.Wait())
 }
 
+func tryRefreshTableErr(err error) bool {
+	errCode, ok := pkgsql.GetSQLErrCode(err)
+	if !ok {
+		return false
+	}
+
+	switch errCode {
+	case infoschema.ErrColumnNotExists.Code():
+		return true
+	}
+
+	return false
+}
+
 func (e *executor) singleExecRetry(ctx context.Context, allDMLs []*DML, safeMode bool, retryNum int, backoff time.Duration) error {
 	for _, dmls := range splitDMLs(allDMLs, e.batchSize) {
 		err := util.RetryContext(ctx, retryNum, backoff, 1, func(context.Context) error {
-			return e.singleExec(dmls, safeMode)
+			execErr := e.singleExec(dmls, safeMode)
+			if execErr == nil {
+				return nil
+			}
+
+			if tryRefreshTableErr(execErr) && e.refreshTableInfo != nil {
+				log.Info("try refresh table info")
+				name2info := make(map[string]*tableInfo)
+				for _, dml := range dmls {
+					name := dml.TableName()
+					info, ok := name2info[name]
+					if !ok {
+						var err error
+						info, err = e.refreshTableInfo(dml.Database, dml.Table)
```

Comment:
There may be problems calling `refreshTableInfo` many times in a short time, If so will it have a big impact?

Reply:
as discuss offline I don't think so.

Reply:
Correct it, the conclusion of our offline discussion is that need some
 testing for it